frontend/src/views/web/challenge/leaderboard.html

```diff
@@ -139,6 +151,81 @@ <h5 class="w-300">Leaderboard</h5>
                         </tr>
                     </tbody>
                 </table>
+                <table ng-if="challenge.isResult && challenge.completeLeaderboard.length && challenge.show_complete_leaderboard" class="centered highlight">
```

Comment:
@Kajol-Kumari you won't need all these changes. You can simply add this `div` below line 52 in leaderboard.html -
```
                        <div class="align-right result-wrn col l3 m4 s5" ng-if="challenge.isChallengeHost">
                            <input ng-checked="challenge.getAllEntries" type="checkbox" id="getAllEntries" ng-model="challenge.getAllEntries"
                                ng-change="challenge.toggleLeaderboard(challenge.getAllEntries)" />
                            <label for="getAllEntries"></label>
                            <span> Show all submissions</span>
                        </div>
```

Reply:
For reference, here the controller code for the above HTML -
```
        vm.getAllEntriesOnPublicLeaderboard = function(phaseSplitId) {
            vm.stopLeaderboard = function() {
                $interval.cancel(vm.poller);
            };
            vm.stopLeaderboard();

            vm.isResult = true;
            vm.phaseSplitId = phaseSplitId;
            // loader for exisiting teams
            vm.isExistLoader = true;
            vm.loaderTitle = '';
            vm.loaderContainer = angular.element('.exist-team-card');

            vm.startLoader("Loading Leaderboard Items");

            // Show leaderboard
            vm.leaderboard = {};
            parameters.url = "jobs/" + "phase_split/" + vm.phaseSplitId + "/public_leaderboard_all_entries/?page_size=1000";
            parameters.method = 'GET';
            parameters.data = {};
            parameters.callback = {
                onSuccess: function(response) {
                    var details = response.data;
                    vm.leaderboard = details.results;
                    for (var i = 0; i < vm.leaderboard.length; i++) {
                        vm.leaderboard[i]['submission__submitted_at_formatted'] = vm.leaderboard[i]['submission__submitted_at'];
                        vm.initial_ranking[vm.leaderboard[i].id] = i + 1;
                        var dateTimeNow = moment(new Date());
                        var submissionTime = moment(vm.leaderboard[i].submission__submitted_at);
                        var duration = moment.duration(dateTimeNow.diff(submissionTime));
                        if (duration._data.years != 0) {
                            var years = duration.asYears();
                            vm.leaderboard[i].submission__submitted_at = years;
                            if (years.toFixed(0) == 1) {
                                vm.leaderboard[i].timeSpan = 'year';
                            } else {
                                vm.leaderboard[i].timeSpan = 'years';
                            }
                        } else if (duration._data.months != 0) {
                            var months = duration.months();
                            vm.leaderboard[i].submission__submitted_at = months;
                            if (months.toFixed(0) == 1) {
                                vm.leaderboard[i].timeSpan = 'month';
                            } else {
                                vm.leaderboard[i].timeSpan = 'months';
                            }
                        } else if (duration._data.days != 0) {
                            var days = duration.asDays();
                            vm.leaderboard[i].submission__submitted_at = days;
                            if (days.toFixed(0) == 1) {
                                vm.leaderboard[i].timeSpan = 'day';
                            } else {
                                vm.leaderboard[i].timeSpan = 'days';
                            }
                        } else if (duration._data.hours != 0) {
                            var hours = duration.asHours();
                            vm.leaderboard[i].submission__submitted_at = hours;
                            if (hours.toFixed(0) == 1) {
                                vm.leaderboard[i].timeSpan = 'hour';
                            } else {
                                vm.leaderboard[i].timeSpan = 'hours';
                            }
                        } else if (duration._data.minutes != 0) {
                            var minutes = duration.asMinutes();
                            vm.leaderboard[i].submission__submitted_at = minutes;
                            if (minutes.toFixed(0) == 1) {
                                vm.leaderboard[i].timeSpan = 'minute';
                            } else {
                                vm.leaderboard[i].timeSpan = 'minutes';
                            }
                        } else if (duration._data.seconds != 0) {
                            var second = duration.asSeconds();
                            vm.leaderboard[i].submission__submitted_at = second;
                            if (second.toFixed(0) == 1) {
                                vm.leaderboard[i].timeSpan = 'second';
                            } else {
                                vm.leaderboard[i].timeSpan = 'seconds';
                            }
                        }
                    }
                    vm.phaseName = vm.phaseSplitId;
                    vm.startLeaderboard();
                    vm.stopLoader();
                    vm.scrollToEntryAfterLeaderboardLoads();
                },
                onError: function(response) {
                    var error = response.data;
                    vm.leaderboard.error = error;
                    vm.stopLoader();
                }
            };

            utilities.sendRequest(parameters);
        };

        if (vm.phaseSplitId) {
            vm.getLeaderboard(vm.phaseSplitId);
        }
        vm.getAllEntries = false;
        vm.toggleLeaderboard = function(getAllEntries){
            vm.getAllEntries = getAllEntries;
            if (vm.phaseSplitId) {
                if (vm.getAllEntries){
                    vm.getAllEntriesOnPublicLeaderboard(vm.phaseSplitId)
                }
                else{
                    vm.getLeaderboard(vm.phaseSplitId);
                }
            }
        }

```